core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -645,19 +665,32 @@ private static void failLateSubscriber(EventExecutor executor,
 
         @Override
         public void abort() {
+            abort(AbortedStreamException::get);
+        }
+
+        @Override
+        public void abort(Throwable cause) {
+            requireNonNull(cause, "cause");
+            abort(() -> cause);
+        }
+
+        @Override
+        public void abort(Supplier<? extends Throwable> causeSupplier) {
```

Comment:
For example, we could define `abort0(@Nullable Supplier causeSupplier)` and use the singleton `AbortingSubscriber` when `causeSupplier` is `null`?

Reply:
That is a good idea. I will address it :-)